File: Phase_1/Extra/RotateArray2.java
import java.util.Arrays;

public class RotateArray2 {
    public void rotate(int[] nums, int k) {
        int N = nums.length;
        k = k % N;

        int[] arr = new int[N];
        arr = Arrays.copyOf(nums, N);
        
        for (int i = 0; i < N; i++) {
            int idx = (i + N - k) % N;
            nums[i] = arr[idx];
        }
    }
}

File: Phase_1/Extra/RotateArray2.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    void rotate(vector<int> &nums, int k)
    {
        int N = nums.size();
        k = k % N;

        vector<int> arr(N);
        arr = nums;

        for (int i = 0; i < N; i++)
        {
            int idx = (i + N - k) % N;
            nums[i] = arr[idx];
        }
    }
};